lib/state.js

```diff
@@ -277,14 +283,23 @@ StateResolver.prototype.isPathInNodeModulesDirectory_ = function(path) {
 };
 
 StateResolver.prototype.resolveFrame_ = function(frame, resolveVars) {
-  var args = resolveVars ? this.resolveArgumentList_(frame) : [{
-    name: 'arguments_not_available',
-    varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
-  }];
+  var args = this.extractArgumentsList_(frame);
   var locals = resolveVars ? this.resolveLocalsList_(frame, args) : [{
     name: 'locals_not_available',
     varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
   }];
+  if (semver.satisfies(process.version, '<1.6')) {
+    args = resolveVars && (args.length > 0) ? this.resolveArgumentList_(args) : [{
+      name: 'arguments_not_available',
+      varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+    }];
+  } else {
+    args = [{
```

Comment:
There needs to be a comment here why the value of `args` provided by `extractArgumentsList_` is being ignored.  If I am reading things correctly, you do not trust the answers provided by Frame#argumentName and Frame#argumentList with Node.js >= 1.6?